tool/tsh/common/proxy_test.go

```diff
@@ -1100,14 +1100,15 @@ func mustSearchEvents(t *testing.T, auth *auth.Server) []apievents.AuditEvent {
 }
 
 func mustFindFailedNodeLoginAttempt(t *testing.T, s *suite, nodeLogin string) {
-	av := mustSearchEvents(t, s.root.GetAuthServer())
-	for _, e := range av {
-		if e.GetCode() == events.AuthAttemptFailureCode {
-			require.Equal(t, e.(*apievents.AuthAttempt).Login, nodeLogin)
-			return
+	require.Eventually(t, func() bool {
+		es := mustSearchEvents(t, s.root.GetAuthServer())
```

Comment:
`mustSearchEvents` is going to call `require.NoError`, which will fail the entire test the first time this fails, even if a future call succeeds. This is probably not what you want. (On top of this, the closure will now run in a different goroutine where it's not legal to call `require.NoError`)

I think `require.EventuallyWithT` and updating `mustSearchEvents` to use `assert` instead of `require` should do the trick.